python/ray/tune/utils/callback.py

```diff
@@ -3,22 +3,99 @@
 
 from ray.tune.callback import Callback
 from ray.tune.syncer import SyncConfig
+from ray.tune.logger import CSVLogger, DEFAULT_LOGGERS, ExperimentLogger, \
+    JsonLogger, LegacyExperimentLogger, Logger
 from ray.tune.syncer import SyncerCallback
 
 
 def create_default_callbacks(callbacks: Optional[List[Callback]],
-                             sync_config: SyncConfig):
+                             sync_config: SyncConfig,
+                             loggers: Optional[List[Logger]]):
 
     callbacks = callbacks or []
+    has_syncer_callback = False
+    has_csv_logger = False
+    has_json_logger = False
 
-    # Check if there is a SyncerCallback
-    has_syncer_callback = any(isinstance(c, SyncerCallback) for c in callbacks)
+    # Track syncer obj/index to move callback after loggers
+    last_logger_index = None
+    syncer_index = None
+
+    if not loggers:
+        # If no logger callback and no `loggers` have been provided,
+        # add DEFAULT_LOGGERS.
+        if not any(
+                isinstance(callback, ExperimentLogger)
+                for callback in callbacks):
+            loggers = DEFAULT_LOGGERS
+
+    # Create LegacyExperimentLogger for passed Logger classes
+    if loggers:
+        # Todo(krfricke): Deprecate `loggers` argument, print warning here.
+        add_loggers = []
+        for trial_logger in loggers:
+            if isinstance(trial_logger, ExperimentLogger):
```

Comment:
When will users pass an ExperimentLogger via `loggers`?

Reply:
They shouldn', but they might, so I guess we should handle this. We could also raise an error here instead